### ISSUE TITLE
Default values on StdImageField and variations

### ISSUE THREAD

**Issue by telemaco** [codingjoe/django-stdimage#195]
Could be really interesting the posibility to define in the model field a set of default values allocated on the static storage. Something like this:

````python
avatar = StdImageField(upload_to=UploadToUUID(path='avatars'),
                       variations={'big': {'width': 200,
                                           'height': 200,
                                           'default_image_path': 'images/default-avatar_200.png'
                                           },
                                   'medium': {'width': 100,
                                              'height': 100,
                                              'default_image_path': 'images/default-avatar_100.png'
                                              },
                                   'small': {'width': 50,
                                             'height': 50,
                                             'default_image_path': 'images/default-avatar_50.png'
                                             }
                                   },
                       default_image_path='images/default-avatar_500.png',
                       blank=True)

````

What do you think?

This repository implements something similar [Django-Default-ImageField](https://github.com/LeeHanYeong/Django-Default-ImageField)

**Comment by codingjoe** [comment]
Hi @telemaco I don't know if I understand you correctly so let me just repeat what I understood.
I read two different questions, if you want to pass attributes to the storage, please just pass a custom storage class. `django.db.FileField` has a `storage` keyword argument.
Also you seem to want the possibility to define a default path. This I would solve using the `upload_to` keyword argument. It also supports callables. If you provide a callable the return value will be treated as the full path to the file.

In other words both features are included in Django itself.

Let me know if I can be of any more help and if you have further questions.

Best
-Joe

**Comment by codingjoe** [comment]
Hi @telemaco I would recommend to build your own template for that and include it where you need it, this way your reduce code redundancy. Something like:
```html
{% load static %}
{% if user.avatar %}
<img src="{{ user.avatar.small.url }}" title="{{ user.username }}" alt="users avatar">
{% else %}
<img src="{% static 'placeholder.png' %}" title="{{ user.username }}" alt="default avatar">
{% endif %}
```
